doc/debug-system.md

```diff
@@ -0,0 +1,391 @@
+} dtm_op_e;
+
+typedef struct packed {
+  logic [6:0]  addr;
+  dtm_op_e     op;
+  logic [31:0] data;
+} dmi_req_t;
+
+output logic     dmi_req_valid_o,
+input  logic     dmi_req_ready_i,
+output dmi_req_t dmi_req_o
+```
+
+#### DMI Response Channel
+
+**Signal** | **Width (bit)** | **Direction**  | **Description**
+---------- | --------------- | -------------- | -------------------------------------------
+resp       | 2               | device -> host | Result of the operation. DTM_SUCCESS = 2'h0
+data       | 32              | device -> host | requested read data
+valid      | 1               | device -> host | handshaking: response is valid
+ready      | 1               | host -> device | handshaking: transaction processed
+
+**SystemVerilog interface definition (host side)**
+
+```verilog
+typedef struct packed  {
+  logic [31:0] data;
+  logic [1:0]  resp;
+} dmi_resp_t;
+
+input  dmi_resp_t dmi_resp_i,
+output logic      dmi_resp_ready_o,
+input  logic      dmi_resp_valid_i
+```
+
+#### Protocol
+
+A DMI transaction consists of a request and a response.
+ A DMI host issues a request on the request channel.
+ The request is transmitted as soon as both ready and valid are high.
+ After some time, the device sends the answers the request on the response channel.
+ The response is considered sent as soon as ready and valid of the response channel are both high.
+ A transaction is completed as soon as the response is received by the host.
+ Only one transaction may be active at any given time.
+
+The following timing diagram shows two DMI transactions.
+
+A READ transaction, reading address 0x12 successfully, the value 0x1234 is returned.
+A WRITE transaction, writing the value 0xabcd to address 0x34 successfully.
+
+![DMI protocol](dmi_protocol.svg)
+
+## System Bus Access
+
+The system bus and all attached peripherals, including the memories, can be accessed from the host system through the System Bus Access (SBA) component.
+A typical use case is writing the program memory through this interface, or verifying its contents.
+The implementation conforms to the [RISC-V Debug Specification v0.13](https://github.com/riscv/riscv-debug-spec/raw/4e0bb0fc2d843473db2356623792c6b7603b94d4/riscv-debug-release.pdf), refer to this document for further information.
+
+## Debug Memory
+
+The Debug Module exposes a 16 kB memory over its device bus interface.
+This memory is called the Debug Memory.
+It consists of a ROM portion, containing the Debug ROM, multiple memory-mapped control and status registers, and a RAM portion, the Program Buffer.
+The Debug Memory should only be accessible from the CPU if it is in debug mode.
+
+### Debug Memory Map
+
+The memory map is an implementation detail of the Debug Module and should not be relied on when using the Debug Module.
+
+Address         | Description
+--------------- | ------------------------------------------------------------------------------------------------------------------------------------------
```

Comment:
I'd make a Description column narrower, so that Addresses are all in one line.

Right now it renders like that:
![image](https://user-images.githubusercontent.com/1504981/66566091-60838600-eb64-11e9-9a65-9a5ad3b69290.png)

As you can see starting from AbstractCmd addresses are two line.

Reply:
That's a limitation of Markdown, there's no way to control the formatting of the table.

Reply:
Ah, yeah. It seems that for GitHub Markdown parser it's not possible. In Pandoc column width is taken as proportion of dashes in the first row. Here's a rather extreme example:
![image](https://user-images.githubusercontent.com/1504981/66581565-9a628580-eb80-11e9-9686-34fd9316cba2.png)

The second one has header as follows:
```
| Address | Description |
|----|-|
```

Reply:
Let's put it like that: I have a reasonably strong dislike for Markdown (exactly for reasons like this), but in this case, going for Markdown and living with however GitHub renders it is the easiest way forward.

Reply:
Yeah, I understand you perfectly :wink: Recently I've been working with Pandoc a little bit, so that's where it came from.